.github/workflows/test.yml

```diff
@@ -0,0 +1,99 @@
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup cache
+        uses: actions/cache@v2
+        with:
+          path: $HOME/SpacemanDMM
+          key: ${{ runner.os }}-spacemandmm
+      - name: Install Tools
+        run: |
+          pip3 install setuptools
+          bash tools/ci/install_build_tools.sh
+          bash tools/ci/install_spaceman_dmm.sh dreamchecker
+          pip3 install -r tools/mapmerge2/requirements.txt
+      - name: Run Linters
+        run: |
+          bash tools/ci/check_filedirs.sh beestation.dme
+          bash tools/ci/check_changelogs.sh
+          find . -name "*.php" -print0 | xargs -0 -n1 php -l
+          find . -name "*.json" -not -path "*/node_modules/*" -print0 | xargs -0 python3 ./tools/json_verifier.py
+          bash tools/ci/build_tgui.sh
+          bash tools/ci/check_grep.sh
+          python3 tools/mapmerge2/dmi.py --test
+          ~/dreamchecker
+  compile_all_maps:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup cache
+        uses: actions/cache@v2
+        with:
+          path: $HOME/BYOND
+          key: ${{ runner.os }}-byond
+      - name: Compile All Maps
+        run: |
+          bash tools/ci/install_byond.sh
+          source $HOME/BYOND/byond/bin/byondsetup
+          python3 tools/ci/template_dm_generator.py
+          bash tools/ci/dm.sh -DCIBUILDING -DCITESTING -DALL_MAPS beestation.dme
+  run_all_tests:
+    runs-on: ubuntu-latest
+    services:
+      mysql:
+        image: mysql:latest
+        env:
+          MYSQL_ROOT_PASSWORD: root
+        ports:
+          - 3306
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup cache
+        uses: actions/cache@v2
+        with:
+          path: $HOME/BYOND
+          key: ${{ runner.os }}-byond
+      - name: Setup database
+        run: |
+          sudo systemctl start mysql
+          mysql -u root -proot -e 'CREATE DATABASE bee_ci;'
+          mysql -u root -proot bee_ci < SQL/beestation_schema.sql
+      - name: Install rust-g
+        run: |
+          sudo dpkg --add-architecture i386
+          sudo apt update || true
+          sudo apt install libssl1.1:i386
+          bash tools/ci/install_rust_g.sh
+      - name: Compile and run tests
+        run: |
+          bash tools/ci/install_byond.sh
+          source $HOME/BYOND/byond/bin/byondsetup
+          bash tools/ci/dm.sh -DCIBUILDING beestation.dme
+          bash tools/ci/run_server.sh
+  test_windows:
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Compile
+        run: pwsh tools/ci/build.ps1
+      - name: Create artifact
+        run: |
+          md deploy
+          bash tools/deploy.sh ./deploy
+      - name: Deploy artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: deploy
+          path: deploy
```

Comment:
```suggestion
```

Creating deploy artifacts from external forks is a terrible idea. It means that users could introduce malicious code into their PR and then GitHub would compile that into an executable deploy package that is tied to our repository.